environ/environ.py

```diff
@@ -970,9 +970,15 @@ def _keep_escaped_format_characters(match):
             m1 = re.match(r'\A(?:export )?([A-Za-z_0-9]+)=(.*)\Z', line)
             if m1:
                 key, val = m1.group(1), m1.group(2)
-                m2 = re.match(r"\A'(.*)'\Z", val)
+                # search for value within quotes, ignore comments after # (outside quotes)
```

Comment:
I paraphrased and reformatted your comment a bit to make it clearer and satisfy the code style.

```suggestion
                # Look for value in quotes, ignore post-# comments
                # (outside quotes)
```